lib/spack/spack/spec.py

```diff
@@ -3514,76 +3656,72 @@ def satisfies(self, other, deps=True, strict=False):
         ):
             return False
         if self.versions and other.versions:
```

Comment:
I think this if is redundant if no version == VersionList(). Note that

all([]) == True

as it should.

Reply:
Done in 9e97a4487017a7a626eb045615ddd50876df5c18